ratis-server/src/main/java/org/apache/ratis/server/impl/PeerConfiguration.java

```diff
@@ -47,8 +47,10 @@
   private final Map<RaftPeerId, RaftPeer> listeners;
 
   static Map<RaftPeerId, RaftPeer> newMap(Iterable<RaftPeer> peers, String name, Map<RaftPeerId, RaftPeer> existing) {
-    Objects.requireNonNull(peers, () -> name + " == null");
     final Map<RaftPeerId, RaftPeer> map = new HashMap<>();
+    if (Objects.isNull(peers)) {
+      return map;
+    }
```

Comment:
Since peers cannot be null but listeners can, let's check listeners outside.
```
   PeerConfiguration(Iterable<RaftPeer> peers, Iterable<RaftPeer> listeners) {
     this.peers = newMap(peers, "peers", Collections.emptyMap());
-    this.listeners = newMap(listeners, "listeners", this.peers);
+    this.listeners = Optional.ofNullable(listeners)
+        .map(l -> newMap(listeners, "listeners", this.peers))
+        .orElseGet(Collections::emptyMap);
   }
```